airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -0,0 +1,321 @@
+    limit = 2000
+
+    def __init__(self, sf_api: Salesforce, pk: str, stream_name: str, schema: dict = None, **kwargs):
+        super().__init__(**kwargs)
+        self.sf_api = sf_api
+        self.pk = pk
+        self.stream_name = stream_name
+        self.schema = schema
+
+    @property
+    def name(self) -> str:
+        return self.stream_name
+
+    @property
+    def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
+        return self.pk
+
+    @property
+    def url_base(self) -> str:
+        return self.sf_api.instance_url
+
+    def path(self, **kwargs) -> str:
+        return f"/services/data/{self.version}/queryAll"
+
+    def next_page_token(self, response: requests.Response) -> str:
+        response_data = response.json()
+        if len(response_data["records"]) == self.limit and self.primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
+            return f"WHERE {self.primary_key} > '{response_data['records'][-1][self.primary_key]}' "
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        selected_properties = self.get_json_schema().get("properties", {})
+
+        # Salesforce BULK API currently does not support loading fields with data type base64 and compound data
+        if self.sf_api.api_type == "BULK":
+            selected_properties = {
+                key: value
+                for key, value in selected_properties.items()
+                if not (("format" in value and value["format"] == "base64") or "object" in value["type"])
+            }
+
+        query = f"SELECT {','.join(selected_properties.keys())} FROM {self.name} "
+        if next_page_token:
+            query += next_page_token
+
+        if self.primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
+            query += f"ORDER BY {self.primary_key} ASC LIMIT {self.limit}"
+
+        return {"q": query}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield from response.json()["records"]
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        if not self.schema:
+            self.schema = self.sf_api.generate_schema(self.name)
+        return self.schema
+
+    def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
+        try:
+            yield from super().read_records(**kwargs)
+        except exceptions.HTTPError as error:
+            error_data = error.response.json()[0]
+            if error.response.status_code == codes.FORBIDDEN and not error_data.get("errorCode", "") == "REQUEST_LIMIT_EXCEEDED":
+                self.logger.error(f"Cannot receive data for stream '{self.name}', error message: '{error_data.get('message')}'")
+            else:
+                raise error
+
+
+class BulkSalesforceStream(SalesforceStream):
+
+    limit = 10000
+    JOB_WAIT_TIMEOUT = 10
```

Comment:
```suggestion
    JOB_WAIT_TIMEOUT_MINS = 10
```
to make it clear this is minutes and not seconds.

check the below variable too.